actors/verifreg/src/lib.rs

```diff
@@ -675,6 +685,148 @@ impl Actor {
             data_cap_removed: removed_data_cap_amount,
         })
     }
+
+    pub fn revoke_expired_allocations<BS, RT>(
+        rt: &mut RT,
+        params: RevokeExpiredAllocationsParams,
+    ) -> Result<(), ActorError>
+    where
+        BS: Blockstore,
+        RT: Runtime<BS>,
+    {
+        // since the alloc is expired this should be safe to publically cleanup
+        rt.validate_immediate_caller_accept_any()?;
+        rt.transaction(|st: &mut State, rt| {
+            let mut allocs = MapMap::from_root(rt.store(), &st.allocations, HAMT_BIT_WIDTH, HAMT_BIT_WIDTH).context_code(ExitCode::USR_ILLEGAL_STATE, "failed to load allocations talbe")?;
+            for alloc_id in params.allocation_ids {
+                let alloc = allocs.get::<Address, AllocationID, Allocation>(params.client, alloc_id)
+                .context_code(ExitCode::USR_ILLEGAL_ARGUMENT,
+                "revoke references allocations id that does not belong to client"
+                )?
+                .context_code(
+                ExitCode::USR_ILLEGAL_ARGUMENT,
+            "revoke references allocations id that does not belong to client"
+                )?;
+                if alloc.expiration > rt.curr_epoch() {
+                    return Err(actor_error!(forbidden, "cannot revoke allocation that has not expired"))
```

Comment:
I'm on the fence about silently continuing here.

Reply:
Let's use the BatchReturn type and report exitcodes so we can do the max work possible without swallowing important errors.